spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,1380 @@
+/// `textDocument/definition` to "Go to definition" from project sources to
+/// dependency sources.
+@jsonRequest("buildTarget/dependencySources")
+operation BuildTargetDependencySources {
+    input: DependencySourcesParams
+    output: DependencySourcesResult
+}
+
+/// The build target dependency modules request is sent from the client to the
+/// server to query for the libraries of build target dependencies that are external
+/// to the workspace including meta information about library and their sources.
+/// It's an extended version of `buildTarget/sources`.
+@jsonRequest("buildTarget/dependencyModules")
+operation BuildTargetDependencyModules {
+    input: DependencyModulesParams
+    output: DependencyModulesResult
+}
+
+/// The build target resources request is sent from the client to the server to
+/// query for the list of resources of a given list of build targets.
+///
+/// A resource is a data dependency required to be present in the runtime classpath
+/// when a build target is run or executed. The server communicates during the
+/// initialize handshake whether this method is supported or not.
+///
+/// This request can be used by a client to highlight the resources in a project
+/// view, for example.
+@jsonRequest("buildTarget/resources")
+operation BuildTargetResources {
+    input: ResourcesParams
+    output: ResourcesResult
+}
+
+/// The build target output paths request is sent from the client to the server to
+/// query for the list of output paths of a given list of build targets.
+///
+/// An output path is a file or directory that contains output files such as build
+/// artifacts which IDEs may decide to exclude from indexing. The server communicates
+/// during the initialize handshake whether this method is supported or not.
+@jsonRequest("buildTarget/outputPaths")
+operation BuildTargetOutputPaths {
+    input: OutputPathsParams
+    output: OutputPathsResult
+}
+
+@jsonNotification("build/taskStart")
+operation OnBuildTaskStart {
+    input: TaskStartParams
+}
+
+/// After a `taskStart` and before `taskFinish` for a `taskId`, the server may send
+/// any number of progress notifications.
+@jsonNotification("build/taskProgress")
+operation OnBuildTaskProgress {
+    input: TaskProgressParams
+}
+
+@jsonNotification("build/taskFinish")
+operation OnBuildTaskFinish {
+    input: TaskFinishParams
+}
+
+/// The compile build target request is sent from the client to the server to
+/// compile the given list of build targets. The server communicates during the
+/// initialize handshake whether this method is supported or not. This method can
+/// for example be used by a language server before `textDocument/rename` to ensure
+/// that all workspace sources typecheck correctly and are up-to-date.
+@jsonRequest("buildTarget/compile")
+operation BuildTargetCompile {
+    input: CompileParams
+    output: CompileResult
+}
+
+@jsonRequest("buildTarget/test")
+operation BuildTargetTest {
```

Comment:
Should this be included in the `BuildServer.operations`?

Reply:
Good catch, fixed